src/main/java/organice/logic/parser/DoneCommandParser.java

```diff
@@ -21,19 +22,35 @@
     public DoneCommand parse(String args) throws ParseException {
         requireNonNull(args);
         String trimmedArgs = args.trim();
+
         String prefixNricString = PREFIX_NRIC.toString();
+        String preficResString = PREFIX_RESULT.toString();
 
         String firstNric;
         String secondNric;
+
+        String result;
         String[] nameKeywords = trimmedArgs.split("\\s+");
 
-        if (nameKeywords[0].startsWith(prefixNricString) && nameKeywords[1].startsWith(prefixNricString)) {
-            firstNric = nameKeywords[0].replaceFirst(prefixNricString, "");
-            secondNric = nameKeywords[1].replaceFirst(prefixNricString, "");
-        } else {
+        try {
+            if (nameKeywords[0].startsWith(prefixNricString)
```

Comment:
Now that I have reviewed `ArgumentMultiMap.java`, you may wish you implement the following:
```
ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NRIC, PREFIX_RESULT);
List<String> allNricValues = argMultimap.getAllValues(PREFIX_NRIC);
List<String> allResultValues = argMultimap.getAllValues(PREFIX_RESULT);

if (allNricValues.size() != 2 && allResultValues.size() != 1) {
    throw new ParseException(
                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, DoneCommand.MESSAGE_USAGE));
}

firstNric = allNricValues.get(0);
secondNric = allNricValues.get(1);
result = allResultValues.get(0);

return new DoneCommand(firstNric, secondNric, result);
```

Reply:
thank you so much and amended. i added exception handlings too